packages/strapi-admin/services/role.js

```diff
@@ -233,8 +233,14 @@ const createRolesIfNoneExist = async ({ createPermissionsForAdmin = false } = {}
   authorPermissions.push(...getDefaultPluginPermissions({ isAuthor: true }));
 
   // assign permissions to roles
-  await strapi.admin.services.permission.assign(editorRole.id, editorPermissions);
-  await strapi.admin.services.permission.assign(authorRole.id, authorPermissions);
+  editorPermissions.forEach(p => {
+    p.role = editorRole.id;
+  });
+  authorPermissions.forEach(p => {
+    p.role = authorRole.id;
+  });
+  await strapi.admin.services.permission.createMany(editorPermissions);
```

Comment:
It would be better to have a add(role, permissions) so you can use it here without dealing with those details